theoplayer/getting-started/01-sdks/09-roku/02-displaying-ads.md

```diff
@@ -0,0 +1,161 @@
+
+### Create your ad descriptions
+
+Ads are added to your content by adding ad sources to your source descriptions that are set on the `source` property of the THEOplayer. Here is an example for VAST ad tags:
+
+```brightscript
+m.player.source = {
+    "title": "My VOD Asset",
+    "live": false,
+    "sources": {
+        "src": "https://example.com/my_vod_asset.m3u8",
+        "type": "application/x-mpegurl"
+    },
+    "ads": [
+        {
+            integration: "csai",
+            sources: "https://exampleadserver.com/preroll.xml",
+            timeOffset: "start"
+        },
+        {
+            integration: "csai",
+            sources: "https://exampleadserver.com/midroll.xml",
+            timeOffset: 30
+        },
+        {
+            integration: "csai",
+            sources: "https://exampleadserver.com/midroll2.xml",
+            timeOffset: 190
+        },
+        {
+            integration: "csai",
+            sources: "https://exampleadserver.com/postroll.xml",
+            timeOffset: "end"
+        }
+    ]
+}
+```
+
+For VMAP ads, the timing is container within the VMAP file itself, so the configuration is simpler:
+
+```brightscript
+m.player.source = {
+    "title": "My VOD Asset",
+    "live": false,
+    "sources": {
+        "src": "https://example.com/my_vod_asset.m3u8",
+        "type": "application/x-mpegurl"
+    },
+    "ads": [
+        {
+            integration: "csai",
+            sources: "https://exampleadserver.com/vmap.xml",
+        }
+    ]
+}
+```
+
+Note that the `timeOffset` property must be omitted for VMAP ads.
+
+You may also specify the timings for VAST ads either by using a timecode or by specifying the percent of the asset that should play before displaying the ad. Timecodes should be in the format `HH:MM:SS`.
+
+```brightscript
+m.player.source = {
+    "title": "My VOD Asset",
+    "live": false,
+    "sources": {
+        "src": "https://example.com/my_vod_asset.m3u8",
+        "type": "application/x-mpegurl"
+    },
+    "ads": [
+        {
+            integration: "csai",
+            sources: "https://exampleadserver.com/midroll.xml",
+            timeOffset: "00:05:30"
+        },
+        {
+            integration: "csai",
+            sources: "https://exampleadserver.com/midroll2.xml",
+            timeOffset: "25%"
+        }
+    ]
+}
+```
+
+#### Mixing VAST and VMAP Cues
+
+Currently, this is not supported. Support may be added for this in the future.
+
+### Listen for ad events
+
+To detect events emitted for the ads, you can call the `addEventListener` method on the Ads API. You can see the list of ad events on [the API reference page]("../../../../../../api-reference/roku/#ads-api"). Note that the method used for the listener needs to be exposed through the interface of the node that owns the listener function.
```

Comment:
Don't use quotes inside Markdown links. Also, consider using [an absolute file path](https://docusaurus.io/docs/markdown-features/links) (relative to the project's root directory) for "faraway" links.
```suggestion
To detect events emitted for the ads, you can call the `addEventListener` method on the Ads API. You can see the list of ad events on [the API reference page](/theoplayer/api-reference/roku.md#ads-api). Note that the method used for the listener needs to be exposed through the interface of the node that owns the listener function.
```